lib/atm.rb

```diff
@@ -3,27 +3,27 @@
 class Atm
   attr_accessor :funds
   STANDARD_PIN = 1234
-  STANDARD_EXP_DATE = "10/17"
+  STANDARD_EXP_DATE = "10/17".freeze
 
-   def initialize
+  def initialize
     @funds = 1000
-   end
+  end
 
-   def withdraw(amount, pin, expiry, account)
-      case
-      when negative_value(amount) then return_error_message(:negative_amount)
-      when !check_pin(pin, account.pin) then return_error_message(:wrong_pin)
-      when check_expiry(expiry, account.exp_date) then return_error_message(:card_expired)
-      when !sufficient_fund_in_atm(amount) then return_error_message (:no_sufficient_fund)
-      when amount % 5 != 0 then return_error_message (:non_rounded_amount)
-
-      else
-        do_transaction(amount, account)
-        respons = { status: true,
-          message: :success,
-          date: Date.today.strftime("%F"),
-          amount: amount,
-          bills: do_bill_count(amount)}
+  def withdraw(amount, pin, expiry, account)
```

Comment:
Metrics/CyclomaticComplexity: Cyclomatic complexity for withdraw is too high. [7/6]<br>Metrics/AbcSize: Assignment Branch Condition size for withdraw is too high. [22.83/15] (http://c2.com/cgi/wiki?AbcMetric)<br>Metrics/MethodLength: Method has too many lines. [15/10] (https://github.com/bbatsov/ruby-style-guide#short-methods)